src/object.c

```diff
@@ -712,6 +712,81 @@ robj *getDecodedObject(robj *o) {
     }
 }
 
+/* We create string object pools for client argv object to avoid memory allocation
+ * and deallocation. The string objects use EMBSTR encoding to avoid memory waste.
+ * As memory allocators are typically aligned to 8 bytes, the size of string object
+ * with EMBSTR encoding will be 24, 32, 40, 48, 56, or 64 bytes. We create pools of
+ * string objects with different sizes, now we have 6 pools to cover all EMBSTR
+ * encoding string object with different sizes. */
+#define CLIENT_ARGV_OBJECT_POOL_NUM 6
+#define CLIENT_ARGV_OBJECT_POOL_INIT_COUNT 16
+#define CLIENT_ARGV_OBJECT_POOL_CAPACITY   256
+
+/* Create client argv object pools and initialize them. */
+clientArgvOjectPool *createClientArgvObjectPools(void) {
+    clientArgvOjectPool *pools = zmalloc(sizeof(clientArgvOjectPool) *
+                                         CLIENT_ARGV_OBJECT_POOL_NUM);
+    for (int i = 0; i < CLIENT_ARGV_OBJECT_POOL_NUM; i++) {
+        clientArgvOjectPool *pool = pools + i;
+        pool->object_size = 4 + i*8;
```

Comment:
The minimum is 4, but the minimum jemalloc allocates is 8, does that mean that at least 4 bytes will be wasted?

Reply:
P.S. I remove some comments since now we have all pools for different sizes ,i updated top comment, and i will update code comments.
>As the header of string object with EMBSTR encoding has the fixed size of 20 bytes, and memory allocators are typically aligned to 8 bytes, the size of string object with EMBSTR encoding will be 24, 32, 40, 48, 56, or 64 bytes, so the significant length for storing data is 4, 12, 20, 28, 36, or 44 bytes. 

here size is alloc size of sds, since calling zmalloc_usable_size is costly and should make sure there is enough room to store data, so alloc size is the limit of the significant length.